crates/sp-domains/src/fraud_proof.rs

```diff
@@ -131,14 +131,20 @@ pub enum VerificationError {
 // TODO: refactor FraudProof
 /// Fraud proof for the state computation.
 #[derive(Debug, Decode, Encode, TypeInfo, PartialEq, Eq, Clone)]
-pub enum FraudProof {
+pub enum FraudProof<Number, Hash> {
     InvalidStateTransition(InvalidStateTransitionProof),
+    InvalidTransaction(InvalidTransactionProof),
+    BundleEquivocation(BundleEquivocationProof<Number, Hash>),
+    ImproperTransactionSortition(ImproperTransactionSortitionProof),
 }
 
-impl FraudProof {
+impl<Number, Hash> FraudProof<Number, Hash> {
     pub fn domain_id(&self) -> DomainId {
         match self {
             Self::InvalidStateTransition(proof) => proof.domain_id,
+            Self::InvalidTransaction(proof) => proof.domain_id,
+            Self::BundleEquivocation(proof) => proof.domain_id,
+            Self::ImproperTransactionSortition(proof) => proof.domain_id,
```

Comment:
maybe `ImproperTransactionOrder` ?

Reply:
Ideally, when producing bundles, all the executors will select the transactions from the transaction pool in such a way the selected transactions won't conflict with the others, i.e., no duplicated transactions in two bundles, so order is not appropriate. I'd also think about `ImproperTransactionSelection`, but Jeremiah suggested this name and it's fine to me as well.